src/sql/sql_provider_datafusion/expr.rs

```diff
@@ -112,6 +113,7 @@ pub fn to_sql_with_engine(expr: &Expr, engine: Option<Engine>) -> Result<String>
                         Ok(format!("TO_TIMESTAMP({})", *value as f64 / 1_000_000.0))
                     }
                 }
+                Some(Engine::MySQL) => Ok(format!("TIMESTAMP({})", value / 1_000_000)),
```

Comment:
`value` should be converted to f64 to keep the precision after seconds in timestamp.
Example: 
![image](https://github.com/user-attachments/assets/2b6d9b4b-c5d6-4c2c-908d-b0d8755783a9)

Reply:
Updated